cartographer_ros/cartographer_ros/msg_conversion.cc

```diff
@@ -146,44 +144,50 @@ sensor_msgs::LaserScan ToLaserScan(
   return msg;
 }
 
-sensor_msgs::PointCloud2 ToPointCloud2Message(
+sensor_msgs::msg::PointCloud2 ToPointCloud2Message(
     const int64 timestamp, const string& frame_id,
     const ::cartographer::sensor::PointCloud& point_cloud) {
   auto msg = PreparePointCloud2Message(timestamp, frame_id, point_cloud.size());
-  ::ros::serialization::OStream stream(msg.data.data(), msg.data.size());
+
+  size_t offset = 0;
+  float * const data = reinterpret_cast<float*>(&msg.data[0]);
```

Comment:
What's happening here is that we need to iterate over the data, one float size at a time, filling in the data.  The code is thus casting us to a float pointer so that we can just do offset++, and move forward the 4 bytes every time.  We can't use a const_cast, both because the data isn't const (we modify it in the below loop), and because we are changing types (from a char pointer to a float pointer).

That all being said, I'm open to other suggestions on how to do this.  This is how I would have done it in C (more-or-less), so I did it that way.  There may be a safer/more C++ way to do it.